addons/mail/static/src/core/common/thread.js

```diff
@@ -182,6 +205,15 @@ export class Thread extends Component {
         this.threadService.fetchMoreMessages(this.props.thread);
     }
 
+    async onClickPreferences() {
+        if (!this.store.user?.user?.id) {
+            return;
+        }
+        const actionDescription = await this.orm.call("res.users", "action_get");
```

Comment:
When is `this.store.user?.user?.id` falsy?

I think it's great for the code to work in all environments. I would not show the clickable text if you cannot change the notification preferences.